hopsworks-common/src/main/java/io/hops/hopsworks/common/featurestore/trainingdatasets/TrainingDatasetFacade.java

```diff
@@ -79,6 +80,42 @@ public TrainingDataset findByIdAndFeaturestore(Integer id, Featurestore features
     }
   }
 
+  /**
+   * Retrieves a list of trainingDataset (different versions) given their name and feature store from the database
+   *
+   * @param name name of the trainingDataset
+   * @param featurestore featurestore of the trainingDataset
+   * @return a single TrainingDataset entity
+   */
+  public List<TrainingDataset> findByNameAndFeaturestore(String name, Featurestore featurestore) {
+    return em.createNamedQuery("TrainingDataset.findByFeaturestoreAndName", TrainingDataset.class)
+          .setParameter("featurestore", featurestore)
+          .setParameter("name", name)
+          .getResultList();
+  }
+
+  /**
+   * Retrieves a training dataset given its name, version and feature store from the database
+   *
+   * @param name name of the trainingDataset
+   * @param version version of the trainingDataset
+   * @param featurestore featurestore of the trainingDataset
+   * @return a single TrainingDataset entity
+   */
+  public Optional<TrainingDataset> findByNameVersionAndFeaturestore(String name, Integer version,
+                                                                    Featurestore featurestore) {
+    try {
+      return Optional.of(
+          em.createNamedQuery("TrainingDataset.findByFeaturestoreAndNameVersion", TrainingDataset.class)
+          .setParameter("featurestore", featurestore)
+          .setParameter("name", name)
+          .setParameter("version", version)
+          .getSingleResult());
+    } catch (NoResultException e) {
+      return Optional.empty();
```

Comment:
@SirOibaf in the featuregroup facade we return null instead of Optional. Shall we change that one to Optional as well?

Reply:
Makes sense.